lib/filter.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+//////////////////////////////
+// contains, is, isNot, doesNotContain, exists, matches
+//
+// Include
+//////////////////////////////
+const addModule = function (results, pkg) {
+  try {
+    results[pkg.package.name] = require(pkg.package.name);
+  }
+  catch (e) {
+    if (!results.hasOwnProperty('__unrequireable')) {
+      results.__unrequireable = [];
+    }
+    results.__unrequireable.push(pkg.package.name);
+  }
+  return results;
+}
+
+module.exports = function ModuleFilter(packages, key) {
+  /**
+    * Contains
+  **/
+  this.containsSync = (search) => {
+    var results = {};
```

Comment:
`results` doesn't need to be defined. The return of your object is what's important.

The chainability of `.map` is important because best practice dictates that we use `.map` in general for operations such as this and similar operations may need that chaining. Just going for code consistency.